src/BeamAdapter/component/controller/InterventionalRadiologyController.inl

```diff
@@ -134,9 +134,29 @@ void InterventionalRadiologyController<DataTypes>::init()
     for(unsigned int i=0; i<m_instrumentsList.size(); i++)
         m_instrumentsList[i]->setControlled(true);
 
-    context->get(m_fixedConstraint);
-    if(m_fixedConstraint==nullptr)
-        msg_error()<<"No fixedConstraint found.";
+    if (!l_fixedConstraint)
+    {
+        typename FixedProjectiveConstraint<DataTypes>::SPtr fixedConstraint{};
+        context->get(fixedConstraint);
+        if (fixedConstraint)
+        {
+            l_fixedConstraint.set(fixedConstraint);
+        }
+        else
+        {
+            msg_error() << "No FixedConstraint found. One will be created on the spot but most likely it will not behave as attended.";
+
+            fixedConstraint = sofa::core::objectmodel::New<FixedProjectiveConstraint<DataTypes>>();
+            context->addObject(fixedConstraint);
+            fixedConstraint->addConstraint(0);
+
+            l_fixedConstraint.set(fixedConstraint);
```

Comment:
what about initialization of the `FixedProjectiveConstraint`? Even if you call it, `FixedProjectiveConstraint::init` can also fail, and there is no automatic mechanism to prevent invalidity of the scene. So, I am in favor of triggering an error without backup.

Reply:
Problem is, if you dont create a fixedconstraint (i.e is null), as the IRC code is... let's say, not really solid 🤔, it will crash later on (not test on pointer here and there etc)
The creation here of FixedConstraint is not really intended to serve as a backup actually ; more as a way to not crash by itself 😅 Thats why there is a msg_error (and not a warning).
Or instead of crashing (if we dont do anything) we can do a `fatal()` or something like that... a graceful exit

And as for the breaking, I agree and I assume that I wont do deprecation and stuff 🙊

Reply:
To me the component should be emit an error and be set in invalid state. 
Then the component state should be checked if it is valid.

Reply:
> To me the component should be emit an error and be set in invalid state. Then the component state should be checked if it is valid.

This is what I did few lines after (commented) but only a few functions in IRC does the componentState testing. 
And I dont really want to update all the functions to add the test 🥸